File: esykart-product/src/main/java/com/webientsoft/esykart/product/service/ProductService.java
package com.webientsoft.esykart.product.service;

import com.webientsoft.esykart.common.model.Status;
import com.webientsoft.esykart.common.model.common.FilterModel;
import com.webientsoft.esykart.common.model.common.PaginatedDataModel;
import com.webientsoft.esykart.common.model.product.ProductModel;
import com.webientsoft.esykart.common.model.product.SkuModel;

public interface ProductService {

    void save(SkuModel model);

    void update(Integer productId,SkuModel model);

    ProductModel find(Integer id);

    PaginatedDataModel search(FilterModel model);

    void changeStatus(Integer id, Status status);

}

File: esykart-store/esykart-store-merchant/src/main/resources/public/app/components/product/js/productService.js
angular.module('altairApp').constant('url', 'products')
.service('ProductService',
[ '$http', 'utils', 'url', function($http, utils, url) {
	var service = {};
	service.save = function (data){
    	return $http.post(url,role);
    };
    service.update = function(id,data){
		return $http.put(url+"/" +roleId ,role);
    };
    
    service.find = function(id,projection){
		return $http.get(url +"/" +id+"?projection="+projection);
    };
    service.findAll = function(page, size,sort,projection){
    	var finalUrl = url+"?projection="+projection;
		finalUrl += "&page="+page;
		finalUrl += "&size="+size;
	 	if(sort!= null && sort.length > 0){
			$(sort).each(function(){
				finalUrl += "&sort=";
				finalUrl += this;
			});
		}
		return $http.get(finalUrl);
    };

    service.search = function(filter,projection){
		return $http.get(url +"/search?projection="+projection,filter);
    };
    
    service.changeStatus = function(id,status){
    	return $http.patch(url +"/" + id + "/changeStatus?status="+status);
    }
    return service;
} ]);
